src/workspace.js

```diff
@@ -227,6 +287,27 @@ module.exports = class Workspace extends Model {
     })
   }
 
+  subscribeToMovedItems () {
+    if (this.movedItemSubscription != null) {
+      this.movedItemSubscription.dispose()
+    }
+    const paneLocations = Object.assign({center: this}, this.docks)
```

Comment:
We think you should add a `getLocation` method to `WorkspaceCenter` so that this code can be written like this:

```js
    for (const paneContainer of this.getPaneContainers()) {
      paneContainer.onDidAddPaneItem(({item}) => {
        if (typeof item.getURI === 'function') {
          const uri = item.getURI()
          if (uri != null) {
            this.previousLocations.save(item.getURI(), paneContainer.getLocation())
          }
        }
      }
    }
```

We don't need to save off the disposables returned from `onDidAddPaneItem`, because the `Workspace` will always outlive its pane containers.